### PR TITLE
Rewrite the escalation service to use group contracts

### PR DESCRIPTION
**Pull request by RDIL**
Now squashed!

### CODE REVIEW

**Comment by moonysolari** [comment]
Waiting for Kaki to get all the latest escalation group definitions and level definitions from official.

**Comment by Kakiking** [comment]
Most escalations are done, our custom escalations still don't have group definitions.

**Comment by moonysolari** [comment]
I got these when trying to use this branch:
```
[12:11:48:713] [Error | unknown] Could not find next contract (5d6f7b13-13be-414f-8010-3e97e5b7b490) in group 41ecf8ce-dfd4-4c08-8f44-52dedc3f089a!
[12:11:48:714] [Error | unknown] Could not find next contract (a4013b71-f772-4e2a-8b35-6511ed132bcf) in group 390ba7b6-de27-464a-b8af-6d0ff54c2aec!
[12:11:48:715] [Error | unknown] Could not find next contract (e1e117f4-cf3e-4538-b085-4a226116f143) in group 70150cd2-ef76-4ba8-80cc-b1e63871b030!
[12:11:48:716] [Error | unknown] Could not find next contract (79d966d9-5756-4a2f-b7e7-468a1a1ac351) in group be567ad3-23f4-4d0b-9d2e-b261ea845ef0!
[12:11:48:716] [Error | unknown] Could not find next contract (c03510ef-1932-4b59-997a-3d75ef6e8e8d) in group d336d894-024a-4cd4-9867-dee7de70ee79!
[12:11:48:717] [Error | unknown] Could not find next contract (19623520-8a89-4cc5-93f1-a4356fda16b2) in group 757fd132-0300-45ec-b5bd-bdd48c543b5c!
[12:11:48:718] [Error | unknown] Could not find next contract (f0d8a930-3581-41c8-b985-31eaeef9b920) in group a5e81878-0eae-4bcf-af9b-9a7e7833f85d!
[12:11:48:720] [Error | unknown] Could not find next contract (cb7d1f3e-996f-4955-9b87-bdc40e4160ee) in group a10472e4-0eb3-451d-814d-38837dd0f407!
[12:11:48:734] [Debug | unknown] Discovered 61 escalation groups.
```

This is because the id of an escalation level is different from what is stored in `Metadata.GroupDefinition.Order` of its group contract. @Kakiking Can you update them all with the official ids and make them consistent? We need to ensure they use their official ids because some of them used to use an arbitrary random UUID, and challenges will break if they don't use their official ids.

**Comment by AnthonyFuller** [comment]
Is that all of them?

**Comment by AnthonyFuller** [comment]
And that's a wrap folks! We're ready for review.

**Review by moonysolari — APPROVED** [review]
LGTM. The escalation level picker needs to be fixed/updated afterwards. Doesn't have to be part of this PR though.

File: components/menuData.ts [hunk 1]
```diff
@@ -1750,10 +1744,7 @@ menuDataRouter.get("/contractcreation/create", (req: RequestWithJwt, res) => {
                             Outfit: {
                                 RepositoryId: km.OutfitRepoId,
                                 Required: true,
-                                IsHitmanSuit: isSuit(
-                                    km.OutfitRepoId,
-                                    req.gameVersion,
-                                ),
+                                IsHitmanSuit: isSuit(km.OutfitRepoId),
```
[the remark this answers is not in the document]
Comment (RDIL) [reply components/menuData.ts:1747]: Previous game versions don't list subcategories, so we have to rely on H3 to be a source of truth.

[the remark this answers is not in the document]
Comment (AnthonyFuller) [reply components/menuData.ts:1747]: "we have to rely on H3 to be a source of truth" talk about trust issues...

**Review by AnthonyFuller — APPROVED** [review]

**Comment by moonysolari** [comment]
Also we need to make sure whether we want to include legacy escalations or not. 

Theoretically they should always be included in H1, but not in H2/H3. Therefore I think we can include them in the code but only show them for H1, and toggle their appearance on H2/H3 with a flag. Choosing to do this will also remove the need for the legacy escalations plugin.

**Comment by AnthonyFuller** [comment]
Just fixed the escalation level picker. Was a simple fix as I just had to change how it accessed the escalation mappings and got the level count.

RE: Legacy Escalations
If we're including them in core Peacock, there's no reason to keep them behind a flag for H2/3 as they will work on those games without any extra intervention, and if we're going to add them to H2/3, we'll have to add the challenges too just to be complete.

**Review by AnthonyFuller — CHANGES_REQUESTED** [review]
Requesting changes just to stop merging before Escalation Codenames has been updated with all the new H3 escalations we've added.

**Review by AnthonyFuller — APPROVED** [review]
Added the new H3 escalations to Escalation Codenames. Should be good to go now.

EDIT: Also fixed some issues with escalation completion. What we're going to do with Legacy Escalations still needs to be decided.

**Comment by moonysolari** [comment]
> What we're going to do with Legacy Escalations still needs to be decided.

I agree that they can be added to all game versions.

**Comment by AnthonyFuller** [comment]
What I just pushed makes the hits category for elusives respect the new Season tag, this now removes The Deceivers from the page when playing in Hitman 2016.

**Pull request merged**